x/subscription/spec/01_concepts.md

```diff
@@ -0,0 +1,21 @@
+<!--
+order: 1
+-->
+
+# Concepts
+
+## Subscription
+
+- Merchant create/stop plans on chain, the price is defined on specified token denomination, users should do nessessary token conversion in other ways, e.g. exchange.
+- User subscribe/unsubscribe to plan.
+- Payments are collected automatically at the start of specified interval, the results of collection are written into block events.
+- Create plan transaction will consume some gases to cover the computational cost of future automatic payment collections.
+- In case of automatic collection fails on some subscribers, the chain won't automatic retry later, merchant can either configure it to automatic cancel these subscribers, or retry collection later manually.
+
+### Intervals
+
+We support specify following types of intervals for plan:
+
+- Fixed seconds interval with offset, like "the beginning of each day offset by UTC+8".
+
+- [TODO] Calender interval, like "first day of each month".
```

Comment:
I'm thinking how this can work with future extensibility -- I guess instead of hardcoding integer, or perhaps have it in protobuf as:
```
message Interval {
  oneof value {
    FixedInterval               fixed                 = 1;
    CalendarInterval              calendar                = 2;
  }
}
```

but perhaps to leave it as it is now -- in the future when calendar intervals are added, one can just use the upgrade module and have the data migration that will convert the stored plans

Reply:
sorry I didn't noticed these comments, they are collapsed by default.
I'll see if I can manage to bring in some calendar support.

Reply:
Support crontab syntax directly, that provides best UX.